### ISSUE TITLE
Spawning logic bugs

### ISSUE THREAD

**Issue by Ultimushadow** [SettingDust/RespawnComplex#2]
Alright, so I found a couple things while going through mod testing

- Beds do not skip night, and instead explode in every dimension, including the overworld, and generate a large error report in the log
- Respawn Anchor are completely unable to be interacted through normal means. You cannot fill them with glowstone blocks, they do not register as a respawn point by the mod, and they do not explode in dimensions other than the nether (though I believe this is intended), and attempting to place a block on the face of one leads to a desync with the client for a moment as the client expects to be able to place a block, same thing happens with glowstone blocks as the Anchor flashes momentarily as if it had gained a charge.
- Dying in the nether or end results in the player being teleported to their death position in the overworld, unless they did not have a spawn point in that dimension when they died
- When beds are set to be accepted spawn locations, they allow the player to interact with them once, but after that they will explode
- The beds show text that indicates they are a saved spawn location, but do not appear on /spawn list and /spawn remove indicates they are not actually there
- This extends to Respawn Anchors, as they can be placed with charges using /setblock and right clicking them gives the same message that they are spawn points, but do not function
- It seems you're generating a config file but not putting anything in it
- Blocks seem to only be registered as actual spawn points if the player directly places it down. Blocks interacted with are not added to the list, cannot be removed as the game cannot find anything, and do not spawn the player there
- The portal in the end will always bring the player to spawn, this is less of a bug and more of an oversight I believe, a spawn point list like my suggestion report from earlier may help avoid this situation. 

I have a log of me loading the game with beds and enderchest added to the block tag (enderchest as a non-vanilla and non-packaged example just to make sure they worked, besides the bug requiring players to place down the block themselves it worked as intended!). You should be able to see a large error message when I place down a bed, ignore the ridiculous amount of errors from C2ME when loading the world
Log: [latest.log](https://github.com/SettingDust/RespawnComplex/files/12606169/latest.log)

I have yet to test modded items as spawn points or modded items that are implemented as spawn points like bed blocks, going to hold off on that until the vanilla implementation is working as intended just in case 😅

**Comment by SettingDust** [comment]
lul. The mojmap is lying. It says it's for spawn. But actually for explode
```
public static boolean canSetSpawn(Level level) {
    return level.dimensionType().bedWorks();
}
```

**Comment by Ultimushadow** [comment]
Lol, not exactly the "bed works" you intended

**Comment by SettingDust** [comment]
Respawn anchor and beds works in 0.3.0

**Comment by SettingDust** [comment]
There isn't an event in fabric that will triggered when breaking bed foot. I'll fix that in next release

**Comment by Ultimushadow** [comment]
Okay, report on the latest update:

- Beds now work as spawn points in the overworld, and explode in the end/nether. However, they still save a spawn point despite exploding and do not remove that spawn point
- Spawn points in the end/nether still attempt to spawn the player in the overworld. The coordinates are correct relative to the spawn point in the intended dimension, but the player still ends up in the overworld
- The end portal attempts to spawn the player at their end spawn point location. I can tell because the coordinates it teleports me to are the same as the point in the end, from the bug above
- Loading the mod for the first time is quite buggy if the player already set a spawn point when the world did not have the mod installed. Not super important but would probably be good to make a note of that for new users
- The game tells the player they do not have a spawn point every time they die, including when they have a valid spawn point in the network

That's what I've found so far!

**Comment by SettingDust** [comment]
I fix the most of them in the new release 0.3.1. Need new testing since the others may invalid that I can't reproduce.
Good luck

**Comment by Ultimushadow** [comment]
I'll test tonight when I'm able to work on my pack and let you know!

**Comment by Ultimushadow** [comment]
Okay, it seems like some issues got moved around, here's what I'm getting:
- Respawn anchor now functions as beds do, activating in the correct dimension and detonating in others
- Beds log two spawn points when activated in /spawn list, and sends the activation message up to 8 times
- Respawn anchor gives activation message 1-6 times when charging
- Right clicking a respawn anchor when unable to charge it (when full or when not holding glowstone) enables vanilla spawn behavior, giving the "respawn point set" message
- Dying in the overworld will instead spawn the player at world spawn, claiming no bed was found
- Dying in the nether/end will spawn the player at the world spawn coordinates, but in the nether/end dimension. I was testing this on a flat world and thus was instantly killed again upon respawning, setting my character into a very glitched state, after which I had to execute tp in the overworld and attempting to load the nether/end again required reloading the world.
- However, dying with the respawn anchor active via the vanilla behavior I mentioned above spawned the player at the anchor coordinates. If the player was in the nether, it was simply just exactly how it works in vanilla. In the overworld or the end, well... 
![2023-09-18_03 23 40](https://github.com/SettingDust/RespawnComplex/assets/35935582/70d3de6c-1097-4a0c-87e5-a8f7444df0e9)
This. On reload, everything was fine and the player ends up just being at the spawn coordinates. I have no idea what caused this but before reloading the player can move around in there like normal, it's really weird, a whole dimension of ghost blocks basically
If I had to guess, I think something about your dimension checking system isn't passing on the information correctly to either the server or the client and to resolve this, the game either freezes them or loaded the dimension it _expects_ to be there because it hasn't been told otherwise. When C2ME is loaded, the chunk the player starts in is clear and loaded properly, when C2ME is not loaded, the entire dimension appears this way until reload. It has to be a world reload, F3+A doesn't work.
Here's the log for the list of mods I was using at the time if it helps, the long list of errors towards the end is me interacting with the glitched blocks (right click made them disappear, left click made a breaking sound but the block replaced itself)
[latest.log](https://github.com/SettingDust/RespawnComplex/files/12645931/latest.log)
This was in single player with a superflat world
